run_on_off_experiment.py

```diff
@@ -0,0 +1,249 @@
+                            agent.test_scores.append(score)
+                            writer.add_scalars("eval", {"score": score}, ep)
+                            writer.add_scalars("training", {"loss": losses[0],
+                                                "loss2": losses[1],
+                                                "loss_actor": losses[2],
+                                                "alpha_loss": losses[3]}, ep)
+
+            # Transition to online
+            print("########## TRANSITION: offline -> online ##########")
+            agent.offline = False
+            agent.cql = False
+            agent.iS = False
+            agent.bc = False
+
+            for ep in range(episodes_on): # online
+                s_curr = env.reset()
+                s_curr = np.reshape(s_curr, (1, n_states))
+                s_curr = s_curr.astype(np.float32)
+                done = False
+                score = 0
+                step = 0
+                while not done:
+                    s_curr_tensor = torch.from_numpy(s_curr)
+                    a_curr_tensor, _ = agent.actor.get_action(s_curr_tensor.to(DEVICE), train=True)
+                    a_curr_tensor = a_curr_tensor.detach()
+                    a_curr = a_curr_tensor.cpu().numpy().flatten()
+
+                    s_next, r, done, _ = env.step(a_curr)
+
+                    s_next = np.reshape(s_next, (1, n_states))
+                    s_next_tensor = torch.from_numpy(s_next)
+                    sample = namedtuple('sample', ['s_curr', 'a_curr', 'reward', 's_next', 'done'])
+                    if step == 500:
+                        print("RAN FOR TOO LONG")
+                        done = True
+
+                    sample.s_curr = s_curr_tensor
+                    sample.a_curr = a_curr_tensor
+                    sample.reward = r
+                    sample.s_next = s_next_tensor
+                    sample.done = done
+
+                    if len(agent.experience_replay_on) < agent.batch_size:
+                        agent.experience_replay_on.append(sample)
+                    else:
+                        agent.experience_replay_on.append(sample)
+                        if ep > exploration_eps:
+                            x_batch = random.sample(agent.experience_replay_on, agent.batch_size)
+                            losses = agent.train(x_batch, ep)
+
+                    s_curr = s_next
+                    score += r
+                    step += 1
+                    if done:
+                        per_ratio_logged_episodes.append(ep + episodes_off)
+                        per_ratio_logged_expert_ratios.append(expert_data_ratio)
+                        print(f"ep:{ep}:################Goal Reached###################", score)
+                        agent.test_scores.append(score)
+                        writer.add_scalars("eval", {"score": score}, ep + episodes_off)
```

Comment:
did you decide to just log it every step for online?  just curious since line 380 from sac_on_off.py seems to be logging it according to log_freq = 10

Reply:
yeah I did, thought it'd be more interesting since we are running fewer iterations

Reply:
ok! sounds good